src/Pulsar.Client/Api/Producer.fs

```diff
@@ -39,34 +42,73 @@ type Producer private (producerConfig: ProducerConfiguration, lookup: BinaryLook
                         then
                             Log.Logger.LogInformation("Resending {0} pending messages", pendingMessages.Count)
                             let mutable sentCount = 0
+
                             while pendingMessages.Count > 0 do
-                                let payload = pendingMessages.Dequeue()
-                                do! clientCnx.Send payload
+                                let pendingMessage = pendingMessages.Dequeue()
+                                do this.Mb.PostAndAsyncReply(fun channel ->
+                                    SendMessage (
+                                        pendingMessage.SequenceId,
+                                        pendingMessage.Payload,
+                                        channel)) |> ignore
                                 sentCount <- sentCount + 1
+
                             Log.Logger.LogInformation("{0} pending messages was sent", sentCount)
                     | _ ->
                         Log.Logger.LogWarning("Connection opened but connection is not ready")
                 | ProducerMessage.ConnectionClosed ->
                     do! connectionHandler.ConnectionClosed()
-                | ProducerMessage.SendMessage (payload, channel) ->
+                | ProducerMessage.SendMessage (sequenceId, payload, channel) ->
+
+                    if pendingMessages.Count <= producerConfig.MaxPendingMessages
+                    then
+                        let pendingMessage = { SequenceId = sequenceId; Payload = payload }
+                        pendingMessages.Enqueue(pendingMessage)
```

Comment:
Since we are enqueuing messages in order the SequenceId and messages generation should also move to mailbox, otherwise concurrent produce calls can break the comparison logic below